src/Action/LoginAction.php

```diff
@@ -92,29 +85,28 @@ public function __construct(
         $this->adminPool = $adminPool;
         $this->templateRegistry = $templateRegistry;
         $this->tokenStorage = $tokenStorage;
-        $this->session = $session;
         $this->translator = $translator;
     }
 
     public function __invoke(Request $request): Response
     {
+        $session = $request->getSession();
+
         if ($this->isAuthenticated()) {
-            $this->session->getFlashBag()->add(
+            $session->getFlashBag()->add(
                 'sonata_user_error',
                 $this->translator->trans('sonata_user_already_authenticated', [], 'SonataUserBundle')
             );
 
             return new RedirectResponse($this->urlGenerator->generate('sonata_admin_dashboard'));
         }
 
-        $session = $request->getSession();
-
         $authErrorKey = Security::AUTHENTICATION_ERROR;
```

Comment:
can AuthenticationUtils be used there? the other bundle, NucleosUserAdminBundle has a fallback in case AuthenticationUtils doesn't exist, but it exist since  2.6 ?

https://github.com/nucleos/NucleosUserAdminBundle/blob/2.3.x/src/Action/LoginAction.php#L152-L176

Reply:
Yes, it can be used, but since it does not seem needed to make all builds pass I prefer to do it on another Pr, can you raise an issue about it so we dont forget?

Reply:
It will be done on: https://github.com/sonata-project/SonataUserBundle/pull/1457